app/assets/javascripts/components/common/wiki_select.jsx

```diff
@@ -33,21 +33,36 @@ const WikiSelect = createReactClass({
     let wikis = [];
     if (this.props.wikis) {
       wikis = this.props.wikis.map((wiki) => {
+        wiki.language = wiki.language || 'www'; // for multilingual wikis language is null
         return {
-          value: wiki,
+          value: JSON.stringify(wiki),
           label: `${wiki.language}.${wiki.project}.org`
         };
       });
     }
 
+    const formatValue = (wiki) => {
+      if (wiki) {
+        if (this.props.multi) {
+          wiki = wiki.map((w) => { return { value: JSON.parse(w.value), label: w.label }; });
+        } else {
+          const value = JSON.parse(wiki.value);
+          wiki = { label: wiki.label, value };
+        }
+        this.props.onChange(wiki);
+      } else {
+        this.props.onChange(this.props.multi ? [] : {});
+      }
+    };
```

Comment:
This seems a little convoluted; is this because of the errors with removing the last selected value from a multi-select? If so, is there a simpler way to handle it?

Reply:
I think this is simple enough. The if block is parsing the JSON back to the object, the reasons of why we do this is given in my previous comment.